rpc.go

```diff
@@ -127,19 +211,35 @@ func (r *rpcClient) doRPCWithRetry(ctx context.Context, address string, msg *amq
 		var link *rpc.Link
 		var rsp *rpc.Response
 		var err error
-		link, err = rpc.NewLink(client, address, opts...)
+		isCachedLink := false
+
+		if len(opts) == 0 {
+			// Can use a cached client since there's nothing unique about this particular
+			// link for this call.
+			// It looks like creating a management link with a session filter is the
```

Comment:
Interesting.  Can this link also be cached?

Reply:
It's trickier than what I was doing for non-sessions because the session filter is targeted to a specific session (ie, not just generic to the entire entity).

I need to do a little more homework for this particular one but I figured defaulting to the old behavior would be fine for now.

Reply:
Makes sense.